src/main/java/org/sonarlint/intellij/config/global/SonarQubeServerMgmtPanel.java

```diff
@@ -0,0 +1,258 @@
+import com.intellij.util.containers.ConcurrentFactoryMap;
+import com.intellij.util.containers.FactoryMap;
+import java.util.LinkedList;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.sonarlint.intellij.util.ResourceLoader;
+
+import javax.swing.*;
+import javax.swing.event.ListSelectionEvent;
+import javax.swing.event.ListSelectionListener;
+import java.awt.*;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class SonarQubeServerMgmtPanel {
+  private static final String LABEL_NO_SERVERS = "No servers";
+  private static final String LABEL_ADD_SERVER = "Add server";
+  private static final String LABEL_UNDEFINED = "<undefined>";
+  private static final String EMPTY_CARD_NAME = "empty";
+  private static final String LIST_ICON = "onde-sonar-16.png";
+
+  // UI
+  private JPanel panel;
+  private Splitter splitter;
+  private JPanel serversPanel;
+  private JBList serverList;
+  private JPanel editorPanel;
+  private JPanel emptyPanel;
+
+  // Model
+  private List<SonarQubeServer> servers = new ArrayList<>();
+  private Consumer<SonarQubeServer> changeListener;
+
+  private final FactoryMap<SonarQubeServer, String> serverIds = new ConcurrentFactoryMap<SonarQubeServer, String>() {
+    private int count;
+    @Override
+    protected String create(SonarQubeServer server) {
+      return Integer.toString(count++);
+    }
+  };
+
+  public SonarQubeServerMgmtPanel(SonarLintGlobalSettings settings) {
+    create();
+    load(settings);
+  }
+
+  public void create() {
+    serverList = new JBList();
+    serverList.getEmptyText().setText(LABEL_NO_SERVERS);
+
+    serversPanel = new JPanel(new BorderLayout());
+
+    ToolbarDecorator toolbarDecorator = ToolbarDecorator.createDecorator(serverList).disableUpDownActions();
+
+    toolbarDecorator.setAddAction(new AnActionButtonRunnable() {
+      @Override
+      public void run(AnActionButton anActionButton) {
+        DefaultActionGroup group = new DefaultActionGroup();
+        group.add(new AddServerAction());
+
+        JBPopupFactory.getInstance()
+          .createActionGroupPopup(LABEL_ADD_SERVER, group, DataManager.getInstance().getDataContext(anActionButton.getContextComponent()),
+            JBPopupFactory.ActionSelectionAid.SPEEDSEARCH, true).show(anActionButton.getPreferredPopupPoint());
+      }
+    });
+
+    toolbarDecorator.setRemoveAction(new AnActionButtonRunnable() {
+      @Override
+      public void run(AnActionButton anActionButton) {
+        SonarQubeServer server = getSelectedServer();
+        if (server == null) {
+          return;
+        }
+
+        CollectionListModel model = (CollectionListModel) serverList.getModel();
+        // it's not removed from serverIds and editorList
+        model.remove(server);
+        servers.remove(server);
+
+        if (model.getSize() > 0) {
+          serverList.setSelectedValue(model.getElementAt(0), true);
+        }
+        else {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Move this "else" on the same line that the previous closing curly brace. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ARightCurlyBraceSameLineAsNextBlockCheck)